core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -428,77 +458,46 @@ private[spark] class BlockManager(
       Option(
         shuffleBlockResolver.getBlockData(blockId.asInstanceOf[ShuffleBlockId]).nioByteBuffer())
     } else {
-      doGetLocal(blockId, asBlockResult = false).asInstanceOf[Option[ByteBuffer]]
-    }
-  }
-
-  private def doGetLocal(blockId: BlockId, asBlockResult: Boolean): Option[Any] = {
-    blockInfoManager.lockForReading(blockId) match {
-      case None =>
-        logDebug(s"Block $blockId was not found")
-        None
-      case Some(info) =>
-        doGetLocal(blockId, info, asBlockResult)
+      blockInfoManager.lockForReading(blockId) match {
+        case None =>
+          logDebug(s"Block $blockId was not found")
+          None
+        case Some(info) =>
+          Some(doGetLocalBytes(blockId, info))
+      }
```

Comment:
don't know if we care about the debug message, but a nicer way to write this is:
```
blockInfoManager.lockForReading(blockId).map { info => doGetLocalBytes(blockId, info) }
```